src/org/labkey/targetedms/model/RawMetricDataSet.java

```diff
@@ -61,42 +61,48 @@ public String getSeriesLabel()
         {
             return seriesLabel;
         }
-        else if (null != modifiedSequence)
-        {
-            return modifiedSequence;
-        }
         else
         {
+            double scale = Math.pow(10, 4);
             StringBuilder modifiedSL = new StringBuilder();
 
-            if (null != customIonName)
+            if (null != modifiedSequence)
             {
-                modifiedSL.append(" ");
-                modifiedSL.append(customIonName);
-                modifiedSL.append(",");
+                modifiedSL.append(modifiedSequence);
             }
-
-            if (null != ionFormula)
+            else
             {
-                modifiedSL.append(" ");
-                modifiedSL.append(ionFormula);
-                modifiedSL.append(",");
+                if (null != customIonName)
+                {
+                    modifiedSL.append(customIonName);
+                    modifiedSL.append(", ");
+                }
+
+                if (null != ionFormula)
+                {
+                    modifiedSL.append(ionFormula);
+                    modifiedSL.append(",");
+                }
+
+                if (null != massMonoisotopic && null != massAverage)
+                {
+                    modifiedSL.append(" [");
+                    modifiedSL.append(Math.round(massMonoisotopic * scale) / scale);
+                    modifiedSL.append("/");
+                    modifiedSL.append(Math.round(massAverage * scale) / scale);
+                    modifiedSL.append("] ");
+                }
             }
 
-            if (null != this.massMonoisotopic && null != this.massAverage)
+            if (null != massAverage)
```

Comment:
Shouldn't this be mz instead of massAverage?

And let's use DecimalFormat instead of Math.round(), which will probably have some of the same floating point problems. Sorry for not catching this last aspect earlier.